app/controllers/users_controller.rb

```diff
@@ -145,7 +148,12 @@ def login
   def extlogin
     if session[:user]
       user = User.find_by_id(session[:user])
-      login_user(user)
+      if user.disabled?
+        render_error('custom_error', :status => :unauthorized, locals: {message: _('User account is disabled, please contact your administrator')})
```

Comment:
This is not right, this is mean to be used by API. What about redirecting to login?

Reply:
Any idea where should we redirect? I'd hope to redirect to the page where all the users initiate the login process, but for `Apache` method, they got signed in right away, right? Any idea? should we redirrect to unauthorized page? WDYT?

Reply:
On giving this a second thought I think, foreman login page would be fine. Although, the user in that case must get a notification about it being disabled.

Just thinking out loud here:
While logging in from apache, I think we use some apache module (not sure), but in the case of Openid Connect we use `mod_auth_openidc` which authenticates the user and stores its value in the cookies. I wonder what we must do about that. As in do we need to clear the browser cache? I am not sure. @tbrisker any suggestions?

Let me know if you have any suggestions.

Reply:
We (@ezr-ondrej and me) think that the best option will be redirect to login page. In this case I need to show error message about disabled user. So redirect to OpenID provider take me the chance to show the error. What do you think @rahulbajaj0509 and @tbrisker ?

Reply:
Go ahead :+1:  Let us not block this PR on the external user redirection page. As @ezr-ondrej mentioned, this PR looks good, once we have the test cases we can go ahead and merge it :)